src/sys/socket/sockopt.rs

```diff
@@ -1015,6 +1039,50 @@ sockopt_impl!(
     libc::IP_TTL,
     libc::c_int
 );
+#[cfg(any(apple_targets, linux_android))]
+#[cfg(feature = "net")]
+sockopt_impl!(
+    /// Retrieve the current time-to-live field for every
+    /// IPv4 packet received on this socket.
+    IpRecvTtl,
+    Both,
+    libc::IPPROTO_IP,
+    libc::IP_RECVTTL,
+    bool
+);
+#[cfg(any(apple_targets, linux_android, target_os = "freebsd",))]
+#[cfg(feature = "net")]
+sockopt_impl!(
+    /// Retrieve the current time-to-live field for every
+    /// IPv4 packet received on this socket.
+    IpTtl,
+    Both,
+    libc::IPPROTO_IP,
+    libc::IP_TTL,
+    libc::c_int
+);
+#[cfg(any(apple_targets, linux_android))]
+#[cfg(feature = "net")]
+sockopt_impl!(
+    /// Retrieve the current hop limit field for every
+    /// IPv6 packet received on this socket.
+    Ipv6RecvHopLimit,
+    Both,
+    libc::IPPROTO_IPV6,
+    libc::IPV6_RECVHOPLIMIT,
+    bool
+);
```

Comment:
Even though this option is not present in the corresponding `ip6` man page, there is [a bug report](https://bugs.freebsd.org/bugzilla/show_bug.cgi?id=197256), which seems to indicate that FreeBSD has this option supported

Reply:
`libc` doesn't seem to export it: https://github.com/nix-rust/nix/pull/2203/checks?check_run_id=20299209049